### ISSUE TITLE
support for `nix run`

### ISSUE THREAD

**Issue by trusktr** [chisui/zsh-nix-shell#17]
Hello! This works great with nix-shell (thanks!), but running `nix run ...` still uses `bash`. Any workaround?

**Comment by chisui** [comment]
`nix run` doesn't support the `NIX_BUILD_SHELL` variable, which this plugin uses as its entry point. In fact they [hardcode `bash`](https://github.com/NixOS/nix/blob/master/src/nix/run.cc#L24).

You could hack something together by temporarily putting a shim named `bash` in your `PATH`, I just haven't done it. The whole nix 2 cli suite was quite immature when I started with nix and so I use the v1 version for most of my needs.

Pull requests welcome ;)

**Comment by chisui** [comment]
It seems that [`nix`](https://github.com/NixOS/nix/blob/9747ea84b43c75f719d719673b9d0a7cb50d34e5/src/nix/run.cc#L64) now evaluates the `SHELL` environment variable.

**Comment by hacker1024** [comment]
> It seems that [`nix`](https://github.com/NixOS/nix/blob/9747ea84b43c75f719d719673b9d0a7cb50d34e5/src/nix/run.cc#L64) now evaluates the `SHELL` environment variable.

This is correct: https://github.com/NixOS/nix/commit/9d04b5da17898fd564308423bc7d10fc929efe18
Introduced in 2.4.
